src/luaclang.c

```diff
@@ -1,14 +1,367 @@
+ */
+static int get_storage_class(lua_State *L)
+{
+        CXCursor cur = to_CXCursor(L, 1);
+        enum CX_StorageClass sc_specifier = clang_Cursor_getStorageClass(cur);
+        lua_pushstring(L, storage_class_str(sc_specifier));
+        return 1;
+}
+
+static int get_result_type(lua_State *L)
+{
+        CXCursor cur = to_CXCursor(L, 1);
+        CXType *type = new_CXType(L);
+        *type = clang_getCursorType(cur);
+        CXString result_type = clang_getTypeSpelling(clang_getResultType(*type));
+        lua_pushstring(L, clang_getCString(result_type));
+        return 1;
+}
+
+/*      
+        Format - cur:getNumArgs()
+        Parameter - cur - Cursor corresponding to which the number non-variadic arguments is to be obtained
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TYPES.html#ga5254f761b57fd78de3ac9c6bfcaa7fed
+*/
+static int get_num_args(lua_State *L)
+{
+        CXCursor cur = to_CXCursor(L, 1);
+        int num_args = clang_Cursor_getNumArguments(cur);
+        lua_pushnumber(L, num_args);
+        return 1;
+}
+
+/*      
+        Format - cur:isFunctionInline()
+        Parameter - cur - Cursor which is to be checked (whether the function declaration `cur` represents is inline)
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TYPES.html#ga5254f761b57fd78de3ac9c6bfcaa7fed
+*/
+static int is_function_inline(lua_State *L)
+{
+        CXCursor cur = to_CXCursor(L, 1);
+        CINDEX_LINKAGE bool is_inline;
+        is_inline = clang_Cursor_isFunctionInlined(cur);
+        lua_pushboolean(L, is_inline);
+        return 1;
+}
+
+static int get_argument(lua_State *L)
+{
+        CXCursor cur = to_CXCursor(L, 1);
+        int arg_index = lua_tointeger(L, 2);
+        CXCursor arg_cursor = clang_Cursor_getArgument(cur, arg_index);
+        CXString arg_cursor_name = clang_getCursorSpelling(arg_cursor);
+        lua_pushstring(L, clang_getCString(arg_cursor_name));
+        clang_disposeString(arg_cursor_name);
+        return 1;
+}
+
+static luaL_Reg cursor_functions[] = {
+        {"getCursorSpelling", get_cursor_spelling},
+        {"getCursorType", get_cursor_type},
+        {"getCursorKind", get_cursor_kind},
+        {"getStorageClass", get_storage_class},
+        {"getResultType", get_result_type},
+        {"getNumArgs", get_num_args},
+        {"isFunctionInline", is_function_inline},
+        {"getArgument", get_argument},
+        {NULL, NULL}
+};
+
+void new_metatable(lua_State *L, const char *name, luaL_Reg *reg)
+{
+        luaL_newmetatable(L, name);
+        lua_pushvalue(L, -1);
+        lua_setfield(L, -2, "__index");
+        luaL_setfuncs(L, reg, 0);
+        lua_pop(L, 1);
+}
+
+int luaopen_luaclang(lua_State *L) 
+{
+        new_metatable(L, INDEX_METATABLE, index_functions);
+        new_metatable(L, TU_METATABLE, tu_functions);
+        new_metatable(L, CURSOR_METATABLE, cursor_functions);
```

Comment:
I assume that the metatable for the Type object will come after this PR, right?

Reply:
Yes